packages/smart-forms-renderer/src/stories/itemTypes/OpenChoice.stories.tsx

```diff
@@ -37,37 +45,182 @@ export default meta;
 type Story = StoryObj<typeof meta>;
 
 // More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
+const clinicCoding = {
+  system: 'http://snomed.info/sct',
+  code: '257585005',
+  display: 'Clinic'
+};
+const targetText = 'Pharmacy';
+
+const qOpenChoiceAnswerOptionBasic = questionnaireFactory([
+  {
+    linkId: 'health-check-location',
+    text: 'Location of health check',
+    type: 'open-choice',
+    extension: [
+      itemControlExtFactory('radio-button'),
+      sdcQuestionnareExtFactory('Other, please specify')
+    ],
+    answerOption: [
+      {
+        valueCoding: clinicCoding
+      }
+    ]
+  }
+]);
+const targetlinkId = 'health-check-location';
+const qrOpenChoiceAnswerOptionBasicResponse = qrFactory([
+  {
+    linkId: targetlinkId,
+    text: 'Location of health check',
+    answer: [
+      {
+        valueString: targetText
+      }
+    ]
+  }
+]);
 
 export const OpenChoiceAnswerOptionBasic: Story = {
   args: {
     questionnaire: qOpenChoiceAnswerOptionBasic
+  },
+  play: async ({ canvasElement }) => {
+    await checkRadioOption(canvasElement, targetlinkId, `label-${clinicCoding.display}`);
+
+    const result = await getAnswers(targetlinkId);
+    expect(result).toHaveLength(1);
+    expect(result[0]).toEqual(expect.objectContaining({ valueCoding: clinicCoding }));
+
+    // Clear value
+    const clearButton = canvasElement.querySelector('button[aria-label="Clear"]');
+    fireEvent.click(clearButton as HTMLElement);
+    // Here we await for debounced store update
+    await new Promise((resolve) => setTimeout(resolve, 500));
+    const resultAfterClear = await getAnswers(targetlinkId);
+    expect(resultAfterClear).toHaveLength(0);
+    const elementAfterClear = await findByLinkId(canvasElement, targetlinkId);
+    const input = elementAfterClear.querySelector('input');
+
+    expect(input).not.toBeChecked();
   }
 };
+const targetOtherLinkid = 'q-item-radio-open-label-box';
+const otherVariantLinkid = 'Other, please specify:';
+const otherTargetText = 'Other variant text';
+
+export const OpenChoiceAnswerOptionBasicOther: Story = {
+  args: {
+    questionnaire: qOpenChoiceAnswerOptionBasic
+  },
+  play: async ({ canvasElement }) => {
+    await checkRadioOption(canvasElement, targetlinkId, `label-${otherVariantLinkid}`);
```

Comment:
it should be     await checkRadioOption(canvasElement, targetlinkId, otherVariantLinkid);